src/formatconvert/main.cpp

```diff
@@ -0,0 +1,113 @@
+#include <QDebug>
+#include <QGuiApplication>
+#include <QCommandLineParser>
+
+#include "mapformat.h"
+#include "tilesetformat.h"
+#include "mapreader.h"
+#include "mapwriter.h"
+
+using namespace Tiled;
+
+template <typename Format>
+Format *findFormat(QString const &fileName) {
+  FormatHelper<Format> formatHelper(FileFormat::ReadWrite, "");
+  foreach (Format *format, formatHelper.formats()) {
+    if (format->supportsFile(fileName))
```

Comment:
I realize that I should probably rename `supportsFile` to `canRead`. It is only meant to find out whether a certain plugin can read an existing file, and not for finding a plugin to write a format with. So when I tested this tool, it failed for me because I was trying to convert a map to a .lua file.